backend/assignments/1/src/main/java/org/example/ExecuteTransaction.java

```diff
@@ -0,0 +1,220 @@
+package org.example;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class ExecuteTransaction  implements Runnable{
+
+    private final static Logger logger = LoggerFactory.getLogger(ExecuteTransaction.class);
+    public static HashMap<String, Trader> hashMapTraders;
+    public static HashMap<String, Coins> hashMapCoins;
+
+    private JsonNode jsonNode;
+
+    public static CountDownLatch countDownLatch;
+    private static final ReentrantLock tradersLock = new ReentrantLock();
+    private static final ReentrantLock coinsLock = new ReentrantLock();
+
+    public ExecuteTransaction(JsonNode jsonNode){
+        this.jsonNode = jsonNode;
+
+    }
+    public synchronized void setHashMapTraders(HashMap<String,Trader> traderHashMap){
```

Comment:
What if user1 needs to work on Coin1 and user2 on coin2.
But user2 will still be in wait state untill the method reopoens lock.
Can the locking be in a better place?